client/app/components/users/UserEdit.jsx

```diff
@@ -0,0 +1,287 @@
+        </Modal>
+        <Button className="w-100 m-t-10" onClick={this.changePassword} data-test="ChangePassword">
+          Change Password
+        </Button>
+      </Fragment>
+    );
+  }
+
+  renderApiKey() {
+    const { user, regeneratingApiKey } = this.state;
+
+    return (
+      <Form layout="vertical">
+        <hr />
+        <Form.Item label="API Key">
+          <InputWithCopy id="apiKey" value={user.apiKey} data-test="ApiKey" readOnly />
+          <Button
+            className="w-100 m-t-10"
+            onClick={this.regenerateApiKey}
+            loading={regeneratingApiKey}
+          >
+            Regenerate
+          </Button>
+        </Form.Item>
+      </Form>
+    );
+  }
+
+  renderPasswordLinkAlert() {
+    const { user, passwordResetLink } = this.state;
+
+    return (
+      <Alert
+        message="Email not sent!"
+        description={(
+          <Fragment>
+            <p>
+              The mail server is not configured, please send the following link
+              to <b>{user.name}</b> to reset their password:
+            </p>
+            <InputWithCopy value={absoluteUrl(passwordResetLink)} readOnly />
+          </Fragment>
+        )}
+        type="warning"
+        className="m-t-20"
+        afterClose={() => { this.setState({ passwordResetLink: null }); }}
+        closable
+      />
+    );
+  }
+
+  renderResendInvitation() {
+    return (
+      <Button
+        className="w-100 m-t-10"
+        onClick={this.resendInvitation}
+        loading={this.state.resendingInvitation}
+      >
+        Resend Invitation
+      </Button>
+    );
+  }
+
+  renderSendPasswordReset() {
+    const { sendingPasswordEmail, passwordResetLink } = this.state;
+
+    return (
+      <Fragment>
+        <Button
+          className="w-100 m-t-10"
+          onClick={this.sendPasswordReset}
+          loading={sendingPasswordEmail}
+        >
+          Send Password Reset Email
+        </Button>
+        {passwordResetLink && this.renderPasswordLinkAlert()}
+      </Fragment>
+    );
+  }
+
+  rendertoggleUser() {
+    const { user, togglingUser } = this.state;
+
+    return user.isDisabled ? (
+      <Button className="w-100 m-t-10" type="primary" onClick={this.toggleUser} loading={togglingUser}>
+        Enable User
+      </Button>
+    ) : (
+      <Button className="w-100 m-t-10" type="danger" onClick={this.toggleUser} loading={togglingUser}>
+        Disable User
+      </Button>
+    );
+  }
+
+  render() {
+    const { user } = this.state;
+
+    return (
+      <div className="col-md-4 col-md-offset-4">
+        <img
+          alt="Profile"
+          src={user.profileImageUrl}
+          className="profile__image"
+          width="40"
+        />
+        <h3 className="profile__h3">{user.name}</h3>
+        <hr />
+        {this.renderBasicInfoForm()}
+        {!user.isDisabled && (
+          <Fragment>
+            {this.renderApiKey()}
+            <h3>Password</h3><hr />
```

Comment:
Whoa too big. I thought similar to:
https://github.com/getredash/redash/blob/389d9ee3a69340e85692f160fb78cf05e48fe969/client/app/components/users/UserEdit.jsx#L170-L171